tensorflow_recommenders_addons/dynamic_embedding/python/kernel_tests/hkv_hashtable_evict_test.py

```diff
@@ -0,0 +1,111 @@
+# limitations under the License.
+# ==============================================================================
+"""unit tests of hkv hashtable ops
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import sys
+import os
+import itertools
+import numpy as np
+
+from tensorflow_recommenders_addons import dynamic_embedding as de
+from tensorflow_recommenders_addons.utils.check_platform import is_windows, is_macos, is_arm64, is_linux, is_raspi_arm
+
+from tensorflow.core.protobuf import config_pb2
+from tensorflow.python.framework import constant_op
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import ops
+from tensorflow.python.framework import test_util
+from tensorflow.python.ops import init_ops
+from tensorflow.python.ops import variable_scope
+from tensorflow.python.ops import variables
+from tensorflow.python.platform import test
+from tensorflow.python.training import server_lib
+from tensorflow.python.client import session
+
+import tensorflow as tf
+try:
+  import tensorflow_io
+except:
+  print()
+
+
+def _type_converter(tf_type):
+  mapper = {
+      dtypes.int32: np.int32,
+      dtypes.int64: np.int64,
+      dtypes.float32: float,
+      dtypes.float64: np.float64,
+      dtypes.string: str,
+      dtypes.half: np.float16,
+      dtypes.int8: np.int8,
+      dtypes.bool: bool,
+  }
+  return mapper[tf_type]
+
+
+default_config = config_pb2.ConfigProto(
+    allow_soft_placement=False,
+    gpu_options=config_pb2.GPUOptions(allow_growth=True))
+
+
+def _get_devices():
+  return ["/gpu:0" if test_util.is_gpu_available() else "/cpu:0"]
+
+
+is_gpu_available = test_util.is_gpu_available()
+
+
+def convert(v, t):
+  return np.array(v).astype(_type_converter(t))
+
+
+class HkvHashtableTest(test.TestCase):
+
+  @test_util.run_in_graph_and_eager_modes()
+  def test_evict_strategy(self):
+    if not is_gpu_available:
+      self.skipTest('Only test when gpu is available.')
+    strategy_i = 0
+    for strategy in de.HkvEvictStrategy:
+      with self.session(use_gpu=True, config=default_config):
+        with self.captureWritesToStream(sys.stderr) as printed:
+          table = de.get_variable(
+              str(strategy),
+              key_dtype=dtypes.int64,
+              value_dtype=dtypes.int32,
+              initializer=0,
+              dim=8,
+              init_size=1024,
+              kv_creator=de.HkvHashTableCreator(
+                  config=de.HkvHashTableConfig(init_capacity=1024,
+                                               max_capacity=1024,
+                                               max_hbm_for_values=1024 * 4 * 8 *
+                                               2,
+                                               evict_strategy=strategy)))
+          self.evaluate(table.size())
+
+          content = "Use Evict Strategy:" + str(strategy_i)
+          self.assertTrue(content in printed.contents())
```

Comment:
I would recommend testing the number of evicted `key-value` at least, about the numberic correctness of eviction, you can choose to trust HKV’s testing quality.

Reply:
I trusted hkv's test, so I only tested whether the evict parameters passed in were initialized correctly. It also feels pointless to test the number of evictions because of uncertainty about when they will occur. Because the number of evictions is also random if the input is random.

Reply:
Make sense